### DISCUSSION TITLE
remoteproc remoteproc0: can't stop rproc: -524

### DISCUSSION THREAD

**Discussion by EricZhangjie** [polarfire-soc/polarfire-soc-documentation#329] [HSS and U-Boot] [answered]
Hello,
Refer to previous discussion: remoteproc remoteproc0: can't stop rproc #304,https://github.com/orgs/polarfire-soc/discussions/304, we try a simple demo that Linux can start and stop the Bare Metal hart application. But if there are some read and write operations to FPGA during initialization of Bare Metal hart application, then stop the bare metal hart will fail. To add, the FPGA was not being accessed at the time of the stop command. Bare Metal hart application can not receive ihc int and also no MIV_RP_MBOX_STOP message received. The terminal will print: "remoteproc remoteproc0: can't stop rproc: -524", what does -524 mean? Where can I find the meaning of the error code.
Another situation: if enter a md command to access the FPGA register on linux, then stop the Bare Metal hart will succeed. Please take a look at the picture below, test1/test2/test3 are with the md command to access the FPGA register, it is ok to stop hart4, test4 without the md command, it failed to stop hart4.
![1fdea76b88cb4c50c5be32f228692709](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/109322517/34fad20f-4df7-4971-8c12-086ed874cd79)

Are there any other documents that provide a detailed introduction of ihc and remoteproc besides the one below?
https://github.com/polarfire-soc/polarfire-soc-documentation/tree/master/applications-and-demos/asymmetric-multiprocessing

**Comment by vfalanis** [comment]
Hi @EricZhangjie , error 524 is a standard error defined in the Linux kernel that means that the operation is not supported, the [errno.h](https://github.com/linux4microchip/linux/blob/linux-5.15-mchp/include/linux/errno.h#L27) header file contains definition for all the error codes. 

Could you please do a `cat /sys/class/remoteproc/remoteproc0/state` to check the state before you do the stop request? I suspect the state is offline before you are requesting the stop in test 4. Could you please confirm this?

Thanks

**Reply by EricZhangjie** [reply]
Hi, i do the test, the state is running before do the stop request in test B. When hart4 is successfully stopped with md command before stop request in test A, the state is offine.
![1685451541851_26BC46A8-ADA1-478c-B88B-C31516F3E77A](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/109322517/7e52f827-0244-4278-930b-928527dfa489)

**Reply by ieexplorer** [reply, marked as answer]
Thanks for your response @vfalanis. We(me & EricZhangjie) have found that the issue is caused by our apb slave implementation in FPGA. The module apb_arbiter.v offered by Microchip requires that the APB_PRDATA stays all'0' when the slave is not accessed. We fixed our slave design and the remoteproc stop worked. Thank for your information.
